src/twisted/web/newsfragments/12277.bugfix

```diff
@@ -0,0 +1 @@
+twisted.internet.address no longer raises DeprecationWarning when used with attrs>=24.1.0. As a side effect, `IPV4Address`/`IPV6Address`/`HostnameAddress` instances are now `frozen` (i.e. immutable).
```

Comment:
I agree that the addresses should be immutable ... but this looks like  a backard incompatible change.

It looks like there is also a performance penalty for enforcing immutable instances.

I think that for this PR, is best to just use `unsafe_hash`.
My understanding is that `unsafe_hash` is just the rename of the old `hash`

I think that `unsafe_hash` was added in `attrs 22.2.0`

Right now we support `attrs >= 21.3.0`

https://github.com/twisted/twisted/blob/bb5ac14f8941ec7d4823e402d4d7d47c663ecc16/pyproject.toml#L39

Reply:
Fair point. I wasn't sure which way is the intended way :) Certainly I don't want to dive in the performance, as I already have too many contexts to keep in the head.

Back to `unsafe_hash` it is, then.